nucypher-core/src/access_control.rs

```diff
@@ -0,0 +1,209 @@
+use alloc::boxed::Box;
+use alloc::string::String;
+use alloc::vec::Vec;
+
+use ferveo::api::{encrypt, Ciphertext, DkgPublicKey, SecretBox};
+use ferveo::Error;
+use serde::{Deserialize, Serialize};
+use umbral_pre::serde_bytes;
+
+use crate::conditions::Conditions;
+use crate::versioning::{
+    messagepack_deserialize, messagepack_serialize, ProtocolObject, ProtocolObjectInner,
+};
+
+/// Authenticated data for encrypted data.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct AuthenticatedData {
+    /// The public key for the encrypted data
+    pub public_key: DkgPublicKey,
+
+    /// The conditions associated with the encrypted data
+    pub conditions: Option<Conditions>,
+}
+
+impl AuthenticatedData {
+    /// Creates a new access control policy.
+    pub fn new(public_key: &DkgPublicKey, conditions: Option<&Conditions>) -> Self {
+        AuthenticatedData {
+            public_key: *public_key,
+            conditions: conditions.cloned(),
+        }
+    }
+
+    /// Return the aad.
+    pub fn aad(&self) -> Box<[u8]> {
+        let public_key_bytes = self.public_key.to_bytes().unwrap();
+        let condition_bytes = self.conditions.as_ref().unwrap().as_ref().as_bytes();
+        let mut result = Vec::with_capacity(public_key_bytes.len() + condition_bytes.len());
+        result.extend(public_key_bytes);
+        result.extend(condition_bytes);
+        result.into_boxed_slice()
+    }
```

Comment:
I'm tempted to suggest this functional approach
```suggestion
	pub fn aad(&self) -> Box<[u8]> {
	    [self.public_key.to_bytes(), self.conditions.as_ref().as_ref().as_bytes()]
	        .concat()
	        .into_boxed_slice()
	}
```

Reply:
@piotr-roslaniec Can you take care of this change for me - either as part of this PR or a follow-up 🙏 ?

Reply:
Sure, we can address any non-critical changes in a follow-up PR. I can take care of that once I test this PR with `nucypher-ts`.